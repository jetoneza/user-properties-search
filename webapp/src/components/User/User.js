import React from 'react';
import PropTypes from 'prop-types';
import { Card, Icon, Image, Grid } from 'semantic-ui-react';

const CardExampleCard = ({ image, name, listingsCount }) => (
  <Grid centered style={{ margin: '1em 0' }}>
    <Card color="blue">
      <Image src={ image } />
      <Card.Content>
        <Card.Header>{ name }</Card.Header>
      </Card.Content>
      <Card.Content extra>
        <a>
          <Icon name='home' />
          { listingsCount } Listing/s
        </a>
      </Card.Content>
    </Card>
  </Grid>
);

CardExampleCard.propTypes = {
  image: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  listingsCount: PropTypes.number.isRequired,
};

export default CardExampleCard;
