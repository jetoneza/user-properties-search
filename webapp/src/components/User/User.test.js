import React from 'react';
import renderer from 'react-test-renderer';
import sinon from 'sinon';

import { mount } from 'enzyme';

import User from './User';

describe('<User /> test', () => {
  const props = {
    image: 'http://image.com',
    name: 'My Name',
    listingsCount: 0,
  };

  test('Should render user', () => {
    const component = <User {...props}/>;
    const wrapper = mount(component);

    const renderedComponent = renderer.create(component);
    const tree = renderedComponent.toJSON();

    expect(tree).toMatchSnapshot();
  });
});


