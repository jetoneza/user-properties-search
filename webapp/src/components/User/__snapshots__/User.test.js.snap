// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<User /> test Should render user 1`] = `
<div
  className="ui centered grid"
  style={
    Object {
      "margin": "1em 0",
    }
  }
>
  <div
    className="ui blue card"
    href={undefined}
    onClick={[Function]}
  >
    <img
      className="ui image"
      src="http://image.com"
    />
    <div
      className="content"
    >
      <div
        className="header"
      >
        My Name
      </div>
    </div>
    <div
      className="extra content"
    >
      <a>
        <i
          aria-hidden="true"
          className="home icon"
        />
        0
         Listing/s
      </a>
    </div>
  </div>
</div>
`;
